### PR TITLE
use GMP_PREFIX

### PR DESCRIPTION
**Pull request by pedritomelenas**
Allow using `$GMP_PREFIX` as explained in https://github.com/gap-packages/NormalizInterface/issues/54#issuecomment-2063525110

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -27,6 +27,25 @@ dnl ##
...
+dnl ##
+
+
+AC_ARG_WITH([gmp],
+  [AS_HELP_STRING([--with-gmp@<:@=PREFIX@:>@],
+    [prefix of GMP installation. e.g. /usr/local])],
+    [],[with_gmp=${GMP_PREFIX}]
+)
+
+GMP_CPPFLAGS="-I$GMP_PREFIX/include"
```
Comment (fingolfin) [inline configure.ac:44]: This won't work if `GMP_PREFIX` is empty. It also completely ignores whatever is passed to the `--with-gmp` command.

The `nq` package contains more sophisticated code to detect GMP.

Comment (pedritomelenas) [reply configure.ac:44]: Right, the `GMP_PREFIX=${GMP_PREFIX}` was giving more issues ... 
I will try to add more tests using the `nq` corresponding file.

Comment (pedritomelenas) [reply configure.ac]: Did some changes, inspired by `np`s `configure.ac`...

Comment (pedritomelenas) [reply configure.ac]: Still fails con cygwin. Sorry.

Comment (pedritomelenas) [reply configure.ac]: Fixed!

File: configure.ac [hunk 2]
```diff
@@ -28,6 +28,40 @@ dnl ## Locate the GAP root dir
...
+GMP_PREFIX=${GMP_PREFIX:-none}
+GMP_CPPFLAGS=""
+GMP_LDFLAGS=""
+
+
+if test "x$GMP_PREFIX" = "xnone" ; then
+    # If a gaproot was specified, try to find GMP in there; otherwise, fall
+    # back to whatever GMP may be found via user specified C/CPP/LDFLAGS
+    if test "${with_gaproot+set}" = set; then
+        echo "with_gaproot = $with_gaproot"
```
Comment (fingolfin) [inline configure.ac:45]:
```suggestion
```

File: configure.ac [hunk 3]
```diff
@@ -28,6 +28,40 @@ dnl ## Locate the GAP root dir
...
+        if test -f ${with_gaproot}/extern/install/gmp/include/gmp.h && test -d ${with_gaproot}/extern/install/gmp/lib ; then
+            echo "adjusting stuff based on gaproot"
```
Comment (fingolfin) [inline configure.ac:47]:
```suggestion
            AC_MSG_RESULT([yes, bundled with GAP at ${with_gaproot}])
```

**Review by fingolfin — COMMENTED** [review]
Thank you so much for working on this @pedritomelenas . I have some comments, perhaps if you find some spare time you'd be willing to address them. Otherwise I'll try to sort this before the next GAP release

**Comment by pedritomelenas** [comment]
Thanks for the suggestions! End of semester here, and lots of grading, corrections. I will work on this as soon as possible.

**Comment by fingolfin** [comment]
No worries. I was just looking at the other packages that should use GMP_PREFIX ... and also just had another look at `nq`, and it seems clear you copied code from there, and several of the things I comment on (e.g. use of `echo`) come from there, so are clear my "fault", not yours. So in particular, I'll also be OK to merge this PR as-is if it works well enough (I'd like to test a little bit / make sure the CI tests cover it). We can always improve it later on.

**Comment by pedritomelenas** [comment]
See https://github.com/gap-packages/NormalizInterface/actions/runs/8799502433
In any case, I will have another look at it during the week. Thanks!

File: prerequisites.sh [hunk 4]
```diff
@@ -68,8 +68,18 @@ else
     #
     # Expert users can help with that, of course, by setting
     # the GMP_DIR variable manually.
+    #
+    # For the moment I am adding it to be $GMP_FREFIX, set in
```
[the remark this answers is not in the document]
Comment (pedritomelenas) [reply prerequisites.sh]: See if the changes are as you expected (feel free to change or revert whatever you think necessary)

**Comment by pedritomelenas** [comment]
All checks passed.

File: prerequisites.sh [hunk 5]
```diff
@@ -124,10 +121,10 @@ if [ "${osname#*CYGWIN}" != "$osname" ]; then
     echo "##"
     # We have to move the Normalize dll to $GAPDIR/.libs, as this is the only
     # place which Cygwin will check for it inside $GAPDIR.
-
     # If using an older GAP built with libtool, put in .libs
-    if test -d ${GAPDIR}/.libs; then cp ${NormalizInstallDir}/bin/cygnormaliz-*.dll ${GAPDIR}/.libs/ ; fi
+    if test -d ${GAPDIR}/.libs; then 
+        cp ${NormalizInstallDir}/bin/cygnormaliz-*.dll ${GAPDIR}/.libs/ ;
```
Comment (fingolfin) [inline prerequisites.sh:126]:
```suggestion
        cp ${NormalizInstallDir}/bin/cygnormaliz-*.dll ${GAPDIR}/.libs/
```

File: Makefile.in [hunk 6]
```diff
@@ -7,8 +7,8 @@ KEXT_SOURCES = src/normaliz.cc
 # workaround for building with GAP 4.11 on Cygwin: we need -std=gnu++11
 # instead of -std=c++11 to avoid an error during compilation; see
 # https://github.com/gap-packages/NormalizInterface/pull/91 for details
-KEXT_CXXFLAGS = @CPPFLAGS@ @NORMALIZ_CPPFLAGS@ -std=gnu++11
-KEXT_LDFLAGS = @LDFLAGS@ -lgmp @NORMALIZ_LDFLAGS@ -lnormaliz -lstdc++
+KEXT_CXXFLAGS = @CPPFLAGS@ @NORMALIZ_CPPFLAGS@  @GMP_CPPFLAGS@ -std=gnu++11
+KEXT_LDFLAGS = @LDFLAGS@ -lgmp @NORMALIZ_LDFLAGS@ @GMP_LDFLAGS@ -lnormaliz -lstdc++
```
Comment (fingolfin) [inline Makefile.in:11]: I am surprised this works, normally `-L` flags must come before `-l` flags they are supposed to affect, i.e., 
```suggestion
KEXT_LDFLAGS = @LDFLAGS@ @GMP_LDFLAGS@ -lgmp @NORMALIZ_LDFLAGS@ -lnormaliz -lstdc++
```

**Pull request merged**